buildpacks/python/tests/build_test.go

```diff
@@ -43,66 +42,92 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
 	when("#Build", func() {
 		var result libcnb.BuildResult
 
-		it.Before(func() {
-			var (
-				appDir string
-				err    error
-			)
-
-			Expect(os.Setenv("BP_FUNCTION", "some_module.some_function"))
-			appDir, cleanupAppDir = tests.SetupTestDirectory(
-				tests.WithFuncYaml(),
-				WithFunctionFile("some_module", "some_function", HTTPFuncTemplate),
-			)
-
-			context = makeBuildContext(
-				withBuildApplicationPath(appDir),
-				withDependencies([]map[string]any{
-					{"id": "invoker-deps", "version": "1.2.3"},
-					{"id": "invoker", "version": "2.3.4"},
-				}),
-				withOptions(map[string]any{
-					"some-option":       "some-value",
-					"some-other-option": "some-other-value",
-				}),
-			)
-
-			result, err = build.Build(context)
-			Expect(err).NotTo(HaveOccurred())
+		when("BP_FUNCTION is valid", func() {
+			it.Before(func() {
+				var (
+					appDir string
+					err    error
+				)
+
+				t.Setenv("BP_FUNCTION", "some_module.some_function")
+
+				appDir, cleanupAppDir = tests.SetupTestDirectory(
+					tests.WithFuncYaml(),
+					WithFunctionFile("some_module", "some_function", HTTPFuncTemplate),
+				)
+
+				context = makeBuildContext(
+					withBuildApplicationPath(appDir),
+					withDependencies([]map[string]any{
+						{"id": "invoker-deps", "version": "1.2.3"},
+						{"id": "invoker", "version": "2.3.4"},
+					}),
+					withOptions(map[string]any{
+						"some-option":       "some-value",
+						"some-other-option": "some-other-value",
+					}),
+				)
+
+				result, err = build.Build(context)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			it("adds expected layers", func() {
+				var layers []string
+				for _, l := range result.Layers {
+					layers = append(layers, l.Name())
+				}
+
+				Expect(layers).To(Equal([]string{
+					"invoker-deps",
+					"invoker",
+					"python-function",
+					"validation",
+				}))
+			})
+
+			it("adds expected labels", func() {
+				Expect(result.Labels).To(Equal([]libcnb.Label{
+					{Key: "some-option", Value: "some-value"},
+					{Key: "some-other-option", Value: "some-other-value"},
+				}))
+			})
+
+			it("adds launch command", func() {
+				Expect(result.Processes).To(Equal([]libcnb.Process{
+					{
+						Type:             "func",
+						Command:          "python",
+						Arguments:        []string{"-m", "pyfunc", "start", "-m", "some_module", "-f", "some_function"},
```

Comment:
Is it expected that this argument for the launch command looks different than in the code?
e.g., `"python", "-m", "pyfunc", "check", "-s", f.applicationPath, "-m", f.module, "-f", f.function`
`check -> start` and no `-s` flag? (I haven't run/dived into this, just curious)

Reply:
Yeah, this is expected. The code you're referring to is for the validation layer, which calls `pyfunc check`. For launch, we want `pyfunc start`. The application path defaults to `.`, which is why I believe it's excluded for the launch command. If you look at the diff a little bit below, you can see that the original start command also didn't set the `-s` flag, so I left as is.